packages/infrastructure/kube_airbyte/main.tf

```diff
@@ -357,6 +366,156 @@ resource "kubernetes_secret" "airbyte_secrets" {
   }
 }
 
+resource "vault_policy" "airbyte" {
+  name = "airbyte-secrets-policy"
+
+  policy = <<-EOT
+    # List secrets
+    path "${local.vault_metadata_path}/*" {
+      capabilities = ["list"]
+    }
+
+    # Read secrets
+    path "${local.vault_data_path}/*" {
+      capabilities = ["read"]
+    }
+
+    # Create and update secrets
+    path "${local.vault_data_path}/*" {
+      capabilities = ["create", "update"]
+    }
+
+    # Delete secrets if needed
+    path "${local.vault_delete_path}/*" {
+      capabilities = ["update"]
+    }
+
+    # Read secret metadata
+    path "${local.vault_metadata_path}/*" {
+      capabilities = ["read"]
+    }
+
+    # Allow Airbyte to check its token
+    path "auth/token/lookup-self" {
+      capabilities = ["read"]
+    }
+
+    # Allow Airbyte to renew its token if needed
+    path "auth/token/renew-self" {
+      capabilities = ["update"]
+    }
+  EOT
+}
+
+resource "vault_token" "airbyte" {
+  policies  = [vault_policy.airbyte.name]
+  renewable = true
+
+  // 768 hours = 32 days
+  ttl       = "768h"
+  period    = "768h"
+
+  metadata = {
+    created_by = "terraform"
+    purpose    = "airbyte_secrets_management"
+  }
+}
+
+resource "kubernetes_secret" "airbyte_vault_token" {
+  metadata {
+    name      = "airbyte-vault-token"
+    namespace = local.namespace
+    labels    = data.pf_kube_labels.labels.labels
+  }
+
+  data = {
+    token = vault_token.airbyte.client_token
+  }
+}
+
+module "vault_token_renewer" {
```

Comment:
This is not the correct way to do this as we have the Vault Secerts Operator for this purpose: https://developer.hashicorp.com/vault/docs/platform/k8s/vso/sources/vault